catalog/ui/search-ui/standard/src/main/webapp/js/widgets/cesium.mapclustering.js

```diff
@@ -0,0 +1,187 @@
+                $.each(this.clusters, function(index, value) {
+                    if(value.points.length > 1) {
+                        var newCluster = {
+                            position : value.position,
+                            point : {
+                                pixelSize : 25,
+                                color : MARKER_COLOR
+                            },
+                            label : {
+                                text : "" + value.points.length,
+                                font : '14px Helvetica',
+                                style : Cesium.LabelStyle.FILL
+                            }
+                        };
+                        value.entity = that.viewer.entities.add(newCluster);
+                    } else {
+                        that.showEntity(value.ref, view);
+                    }
+                });
+            }
+        },
+        showEntity : function(billboard, view) {
+            billboard.show = true;
+            if(typeof view.polygons !== "undefined" ) {
+                var polygons = view.polygons;
+                $.each(polygons, function(index, value) {
+                    value.fill.show = true;
+                    value.outline.show = true;
+                });
+            }
+        },
+        hideEntity : function(billboard, view) {
+            billboard.show = false;
+            if(typeof view.polygons !== "undefined" ) {
+                var polygons = view.polygons;
+                $.each(polygons, function(index, value) {
+                    value.fill.show = false;
+                    value.outline.show = false;
+                });
+            }
+        },
+        fitsInExistingCluster: function(position) {
+            var that = this;
+            var cluster = false;
+            $.each(this.clusters, function(index, value) {
+                if(that.isInRadius(that.getRadius(value), position, value)) {
+                    cluster = value;
+                    return false;
+                }
+            });
+            return cluster;
+        },
+        createNewCluster : function (billboard, view) {
+            var points = [];
+            var views = [];
+            points.push(billboard.position);
+            views.push(view);
+            this.clusters.push({ref : billboard, views : views, position : billboard.position, points: points, radius : DEFAULT_PIXEL_DISTANCE});
+        },
+        addPointToCluster: function(cluster, billboard, view) {
+            var points = cluster.points;
+            var views = cluster.views;
+            points.push(billboard.position);
+            views.push(view);
+            var boundingSphere = Cesium.BoundingSphere.fromPoints(points);
+            var center = boundingSphere.center;
+            cluster.position = center;
+        },
+        removeClusters : function () {
+            var that = this;
+            $.each(this.clusters, function(index, cluster) {
```

Comment:
Instead of using that, _.each can take a context so that you can pass `this` as the context and then `this` will be correct while iterating.  



http://underscorejs.org/#each